beater/pub.go

```diff
@@ -143,8 +142,12 @@ func (p *publisher) processPendingReq(req pendingReq) {
 		defer tx.End()
 	}
 
+	var events []beat.Event
 	span := tx.StartSpan("Transform", "Publisher", nil)
-	events := req.payload.Transform(req.config)
+	for _, transformable := range req.transformables {
+		events = append(events, transformable.Transform(&req.tcontext)...)
```

Comment:
I initially did it to avoid unnecessarily acquiring the lock: https://github.com/elastic/beats/blob/35a7882f69d6daf9ab72f36c6509a00aaa2904d0/libbeat/publisher/pipeline/client.go#L50-L64

but since there's no other threads accessing that lock there shouldn't be any lock contention, so i think it's fine to just use `p.client.Publish`

Reply:
We do instantiate the number of publishers based on available processors, see https://github.com/elastic/apm-server/blob/master/beater/pub.go#L89. 
It would be nice to have some benchmarks for the different versions.

Reply:
@roncohen I saw that you changed the handling now - out of curiosity - did you do some performance measurements?